### PR TITLE
[feature][KLD-215]: configure app with `dex-config.json`

### PR DESCRIPTION
**Pull request by 0x009922**
In fact, PR's changes are described in changed `docs/deploy.md` file. Please consult it.

### CODE REVIEW

File: docs/deploy.md [hunk 1]
```diff
@@ -29,11 +29,30 @@ To build Klaytn-DEX, you need Node.js v16 and PNPM v7 installed:
   npm i -g pnpm
   ```
 
-## Configuration via Environment Variables
+## Configuration
```
Comment (0x009922) [inline docs/deploy.md:32]: @outoftardis, please take a look.

File: docs/modules.md [hunk 2]
```diff
@@ -43,9 +43,6 @@ The `core` contains:
 - domain logic
 - classes for working with Kaikas and Metamask wallets via `caver` and `ethers` providers.
 - ABI (application binary interface) for Klaytn-DEX smart contracts and their methods
```
Comment (0x009922) [inline docs/modules.md:45]: @outoftardis, please take a look.

File: etc/vite-plugin-config.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+      type: 'object',
+      properties: {
+        exchange: SCHEMA_URI,
+        farming: SCHEMA_URI,
+        staking: SCHEMA_URI,
+        snapshot: SCHEMA_URI,
+      },
+      required: ['exchange', 'farming', 'staking', 'snapshot'],
+    },
+    // FIXME should it be a URI?
```
Comment (0x009922) [inline etc/vite-plugin-config.ts:81]: @RDMStreet, please take a look.

File: docs/deploy.md [hunk 4]
```diff
@@ -29,11 +29,30 @@ To build Klaytn-DEX, you need Node.js v16 and PNPM v7 installed:
...
+
+App configuration is split into 2 parts: main configuration file and environment variables.
```
Comment (outoftardis) [inline docs/deploy.md:34]:
```suggestion
App configuration is split into two parts: main configuration file and environment variables.
```

File: docs/deploy.md [hunk 5]
```diff
@@ -29,11 +29,30 @@ To build Klaytn-DEX, you need Node.js v16 and PNPM v7 installed:
...
+
+### Main configuration file
+
+Most of application configuration should be defined in `dex-config.json` file. **This configuration is mandatory.** Also, its contents is strictly validated before being passed into application.
+
+The main configuration includes:
```
Comment (outoftardis) [inline docs/deploy.md:40]:
```suggestion
The main configuration in `dex-config.json` includes:
```

File: docs/deploy.md [hunk 6]
```diff
@@ -29,11 +29,30 @@ To build Klaytn-DEX, you need Node.js v16 and PNPM v7 installed:
...
+
+- tokens whitelist + native and dex token
```
Comment (outoftardis) [inline docs/deploy.md:42]:
```suggestion
- native and DEX tokens
- whitelisted tokens
```

File: docs/deploy.md [hunk 7]
```diff
@@ -29,11 +29,30 @@ To build Klaytn-DEX, you need Node.js v16 and PNPM v7 installed:
...
+- network configuration
+- addresses of smart contracts deployed in the network
+- URIs of subgraphs
+
+`dex-config.example.json` contains a sample configuration, also used by Soramitsu in dev environment.
```
Comment (outoftardis) [inline docs/deploy.md:47]:
```suggestion
The `dex-config.example.json` file contains a sample configuration used by Soramitsu in dev environment.
```

File: docs/deploy.md [hunk 8]
```diff
@@ -29,11 +29,30 @@ To build Klaytn-DEX, you need Node.js v16 and PNPM v7 installed:
...
+
+For more details of configuration schema please consult [`etc/vite-plugin-config.ts`](../etc/vite-plugin-config.ts).
```
Comment (outoftardis) [inline docs/deploy.md:49]:
```suggestion
For more details of configuration schema, consult [`etc/vite-plugin-config.ts`](../etc/vite-plugin-config.ts).
```

File: docs/deploy.md [hunk 9]
```diff
@@ -29,11 +29,30 @@ To build Klaytn-DEX, you need Node.js v16 and PNPM v7 installed:
...
   ```
 
-## Configuration via Environment Variables
+## Configuration
+
+App configuration is split into 2 parts: main configuration file and environment variables.
+
+### Main configuration file
+
+Most of application configuration should be defined in `dex-config.json` file. **This configuration is mandatory.** Also, its contents is strictly validated before being passed into application.
```
Comment (outoftardis) [inline docs/deploy.md:38]:
```suggestion
Most of application configuration should be defined in `dex-config.json` file. **This configuration is mandatory.** Its content is strictly validated before being passed into application.
```

**Review by outoftardis — APPROVED** [review]

**Pull request merged**